src/module.c

```diff
@@ -1535,8 +1595,65 @@ int RM_ReplyWithEmptyArray(RedisModuleCtx *ctx) {
     return REDISMODULE_OK;
 }
 
+/* Reply to the client with an empty map.
+ *
+ * The function always returns REDISMODULE_OK. */
+int RM_ReplyWithEmptyMap(RedisModuleCtx *ctx) {
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return REDISMODULE_OK;
+    addReply(c,shared.emptymap[c->resp]);
```

Comment:
I tend to agree. it may be nice inside of redis, and might have been nice when there was just one type (array), but with all the types now, looking at the API it doesn't make sense.

Reply:
agree, let's drop the new `*Empty*` APIs and have the optimisation to call shared.*empty